### PR TITLE
add "show" and "add" subcommands

### PR DESCRIPTION
**Pull request by joergsteffens**
Added the subcommands "add" to add a completed record and "show" to list records.
Improve argument handling, by using argparser.

### CODE REVIEW

File: timetagger_cli/__main__.py [hunk 1]
```diff
@@ -10,78 +12,99 @@
...
+    parser = subparsers.add_parser(func.__name__, help=func.__doc__.strip())
+    parser.set_defaults(func=func)
+    return parser
+
+
+def setup_parser():
+    """setup argument parsing"""
+    argparser = argparse.ArgumentParser(
+        prog="timetagger", description=timetagger_cli.__doc__.strip()
+    )
```
Comment (almarklein) [inline timetagger_cli/__main__.py:33]: Could you also update the `docs` section in the readme? It just shows the result of calling `timetagger_cli` from the command line.

**Review by almarklein — COMMENTED** [review]
🚀 Thanks, very nice additions. I have just a few comments.

File: timetagger_cli/core.py [hunk 2]
```diff
@@ -135,7 +142,36 @@ def start(description):
...
+def add(args):
+    """Add already finished task."""
+    description = args.description
+    now = int(time.time())
+
+    date = datetime.date.today()
+    if args.date:
+        date = args.date
+
+    start = datetime.datetime.combine(date, args.start_time)
```
[the remark this answers is not in the document]
Comment (joergsteffens) [reply timetagger_cli/core.py:154]: Thankfully argparse takes care of this already:

```
$ python -m timetagger_cli add
usage: timetagger add [-h] [--date DATE] start_time end_time description
timetagger add: error: the following arguments are required: start_time, end_time, description
$ python -m timetagger_cli add test
usage: timetagger add [-h] [--date DATE] start_time end_time description
timetagger add: error: argument start_time: invalid fromisoformat value: 'test'
```

File: timetagger_cli/__main__.py (hunk 1 above)
Comment (joergsteffens) [reply timetagger_cli/__main__.py]: done

**Comment by joergsteffens** [comment]
I now also adapted the tests to work with the new code.

**Review by almarklein — COMMENTED** [review]
I like these changes! CI says the code needs a `black .`, and also found some test failures.

**Comment by joergsteffens** [comment]
Updated by PR. Hope it works now also for Python <= 3.8

**Comment by almarklein** [comment]
On Python 3.6: `AttributeError: type object 'datetime.date' has no attribute 'fromisoformat'`.

If this is tricky to fix, we can also decide to bump the minimal required Python version to 3.7.

**Comment by joergsteffens** [comment]
I adapted it for Python <= 3.6, however only tested it on my machine with a current version of Python. Hopefully the test do run through now.

**Comment by almarklein** [comment]
Sorry for the late reply, the email notification from your last comment got buried in my inbox.  Thanks for this nice contribution!

**Pull request merged**